### PR TITLE
Issue 48538: Authentication provider list respects new "allowInsert" property

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
See [issue](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=48538) for more details.

#### Related Pull Requests
* N/A

#### Changes
* Reset secondary providers after addition or deletion of authconfig
* Disable menu item if allowInsert is false, add hacky css tooltip to avoid disabling-element-hover-element interaction
* Update snapshots

### CODE REVIEW

File: core/src/client/AuthenticationConfiguration/AuthenticationConfiguration.tsx [hunk 1]
```diff
@@ -230,22 +251,28 @@ export class App extends PureComponent<{}, Partial<State>> {
...
             ),
             success: Utils.getCallbackWrapper(() => {
-                this.setState(state => ({
-                    [configType]: state[configType].filter(auth => auth.configuration !== configuration),
-                }));
+                this.setState(
+                    state => ({
+                        [configType]: state[configType].filter(auth => auth.configuration !== configuration),
+                    }),
+                    () => this.resetSecondaryProviders()
```
Comment (labkey-alan) [inline core/src/client/AuthenticationConfiguration/AuthenticationConfiguration.tsx]:
```suggestion
                    this.resetSecondaryProviders
```

File: core/src/client/AuthenticationConfiguration/AuthenticationConfiguration.tsx [hunk 2]
```diff
@@ -230,22 +251,28 @@ export class App extends PureComponent<{}, Partial<State>> {
...
+        this.setState(
+            state => {
+                const prevState = state[configType];
+                const newState = addOrUpdateAnAuthConfig(config, prevState, configType);
+
+                // Update our dirtiness information with added modal, since dirtiness should only track reordering
+                const dirtinessData = { ...state.dirtinessData, [configType]: newState };
+                return { [configType]: newState, dirtinessData };
+            },
+            () => this.resetSecondaryProviders()
```
Comment (labkey-alan) [inline core/src/client/AuthenticationConfiguration/AuthenticationConfiguration.tsx]:
```suggestion
            this.resetSecondaryProviders
```

**Review by labkey-alan — APPROVED** [review]
Looks good, left some very minor suggested changes. I did not manually test.

**Review by labkey-adam — APPROVED** [review]

**Comment by labkey-adam** [comment]
Thanks!

**Pull request merged**